google/cloud/storage/client_sign_url_test.cc

```diff
@@ -107,6 +107,17 @@ TEST_F(CreateSignedUrlTest, V2SignRemote) {
   EXPECT_THAT(*actual, HasSubstr(expected_signed_blob_safe));
 }
 
+/// @test Verify that CreateV2SignedUrl() respects the custom endpoint.
+TEST_F(CreateSignedUrlTest, V2SignCustomEndpoint) {
+  std::string custom_endpoint = "https://storage.mydomain.com";
+
+  Client client(Options{}.set<RestEndpointOption>(custom_endpoint));
+  StatusOr<std::string> actual =
+      client.CreateV2SignedUrl("GET", "test-bucket", "test-object");
+  ASSERT_STATUS_OK(actual);
+  EXPECT_THAT(*actual, HasSubstr(custom_endpoint));
```

Comment:
Maybe `StartsWith` would be a better condition?  And nowadays we would write:

```suggestion
  EXPECT_THAT(actual, IsOkAndHolds(StartsWith(custom_endpoint)));
```

Reply:
This would require some new `using ...` declarations at the top of the file.